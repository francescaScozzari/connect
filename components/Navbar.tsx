import React from 'react'
import styled from 'styled-components'

import { ThemeSwitch } from '@/components/ThemeSwitch'

const Navbar = () => {
  return (
    <Nav>
      <ThemeSwitch />
    </Nav>
  )
}

const Nav = styled.nav`
  width: 100%;
  background-color: ${({ theme }) => theme.colors.neutrals[100]};
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 5px;
`

export { Navbar }
