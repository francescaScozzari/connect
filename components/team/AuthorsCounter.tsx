import React from 'react'
import { styled } from 'styled-components'
import { IconGroup } from '@/components/commons/Icons'

const AuthorsCounter = ({ counter }: { counter: number }) => {
  return (
    <Label>
      <IconGroup title="group" />
      <Bold as="span">{counter}</Bold> researchers identified
    </Label>
  )
}

const Label = styled.span`
  gap: 1em;
  font-size: 0.875rem;
  color: white;
`

const Bold = styled(Label)`
  margin-left: 0.5em;
  font-weight: 700;
`

export { AuthorsCounter }
