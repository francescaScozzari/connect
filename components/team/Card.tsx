import React from 'react'
import * as Accordion from '@radix-ui/react-accordion'

import { AccordionHeader } from './accordion/AccordionHeader'
import { AccordionContent } from './accordion/AccordionContent'

const Card = () => {
  return (
    <Accordion.Root type="single" defaultValue="item-1" collapsible>
      <Accordion.Item value="item-1">
        <AccordionHeader />
        <AccordionContent>
          Yes. It adheres to the WAI-ARIA design pattern.
        </AccordionContent>
      </Accordion.Item>

      <Accordion.Item value="item-2">
        <AccordionHeader>Is it unstyled?</AccordionHeader>
        <AccordionContent>
          Yes. It's unstyled by default, giving you freedom over the look and
          feel.
        </AccordionContent>
      </Accordion.Item>

      <Accordion.Item value="item-3">
        <AccordionHeader>Is it unstyled?</AccordionHeader>
        <AccordionContent>
          Yes. It's unstyled by default, giving you freedom over the look and
          feel.
        </AccordionContent>
      </Accordion.Item>

      <Accordion.Item value="item-4">
        <AccordionHeader>Is it unstyled?</AccordionHeader>
        <AccordionContent>
          Yes. It's unstyled by default, giving you freedom over the look and
          feel.
        </AccordionContent>
      </Accordion.Item>

      <Accordion.Item value="item-5">
        <AccordionHeader>Is it unstyled?</AccordionHeader>
        <AccordionContent>
          Yes. It's unstyled by default, giving you freedom over the look and
          feel.
        </AccordionContent>
      </Accordion.Item>

      <Accordion.Item value="item-6">
        <AccordionHeader>Is it unstyled?</AccordionHeader>
        <AccordionContent>
          Yes. It's unstyled by default, giving you freedom over the look and
          feel.
        </AccordionContent>
      </Accordion.Item>

      <Accordion.Item value="item-7">
        <AccordionHeader>Is it unstyled?</AccordionHeader>
        <AccordionContent>
          Yes. It's unstyled by default, giving you freedom over the look and
          feel.
        </AccordionContent>
      </Accordion.Item>
    </Accordion.Root>
  )
}

export { Card }
