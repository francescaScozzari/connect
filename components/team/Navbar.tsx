import React from 'react'
import styled from 'styled-components'
import Link from 'next/link'

import { InputWithErrors } from '@/components/commons/Input'
import { H3 } from '@/components/commons/Typography'
import { useAppSelector } from '@/store'

const Navbar = () => {
  const { q } = useAppSelector(state => state.search)

  return (
    <Nav>
      <H3.Normal>
        <Link href="/">Lorem Ipsum</Link>
      </H3.Normal>
      <InputWithErrors placeholder={q} />
    </Nav>
  )
}

const Nav = styled.nav`
  width: 100%;
  background-color: ${({ theme }) => theme.colors.neutrals[100]};
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 2em 3em;

  a {
    color: black;
    text-decoration: none;
  }

  div {
    margin: 0 auto;
  }

  @media (max-width: 768px) {
    div {
      display: none;
    }
  }
`

export { Navbar }
