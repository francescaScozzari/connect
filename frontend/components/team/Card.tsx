import React from 'react'
import * as Accordion from '@radix-ui/react-accordion'

import { AccordionHeader } from './accordion/AccordionHeader'
import { AccordionContent } from './accordion/AccordionContent'
import { Author } from '@/models/Authors'

type CardProps = {
  authors: Author[]
}

const Card = ({ authors }: CardProps) => {
  return (
    <Accordion.Root type="single" defaultValue="item-0" collapsible>
      {authors.map((author, index) => {
        return (
          <Accordion.Item key={index} value={`item-${index}`}>
            <AccordionHeader
              fullName={author.fullName}
              university={author.university}
              orcid={author?.orcid}
            />
            <AccordionContent documents={author.documents} />
          </Accordion.Item>
        )
      })}
    </Accordion.Root>
  )
}

export { Card }
