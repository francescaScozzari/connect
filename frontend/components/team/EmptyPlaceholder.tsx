import React from 'react'
import { styled } from 'styled-components'
import Link from 'next/link'

import { IconNoGroup } from '@/components/commons/Icons'
import { H2 } from '../commons/Typography'
import { BackButton } from '../commons/BackButton'
import { SearchTips } from '../commons/SearchTips'
import { AuthorsCounter } from './AuthorsCounter'

const EmptyPlaceholder = () => {
  return (
    <MainSection>
      <AuthorsCounter counter={0} />
      <Container>
        <IconNoGroup title="no team" />
        <H2.Normal color="#2B2D42">Your search returned no results</H2.Normal>
        <Link href="/">
          <BackButton />
        </Link>
        <SearchTips />
      </Container>
    </MainSection>
  )
}

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  gap: 2em;
  padding: 4em;
  border-radius: 20px;
  background-color: white;
`

const MainSection = styled.section`
  width: 100%;
  padding: 4em;
  display: flex;
  flex-direction: column;
  justify-content: start;
  gap: 1em;
  background-color: ${({ theme }) => theme.colors.primary[0]};
`

export { EmptyPlaceholder }
