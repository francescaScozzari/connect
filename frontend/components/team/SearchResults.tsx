import React from 'react'
import { styled } from 'styled-components'

import { Text } from '@/components/commons/Typography'
import { Card } from './Card'
import { AuthorsCounter } from './AuthorsCounter'
import type { Author } from '@/models/Authors'
import { highlightSentence } from '@/utils/utils'

type Props = {
  authors: Author[]
  givenSentence: { text: string; highlights: string[] }
}

const SearchResults = ({ authors, givenSentence }: Props) => {
  const sentence =
    highlightSentence(givenSentence.text, givenSentence.highlights) ?? ''

  return (
    <MainSection>
      <Sidebar>
        <AuthorsCounter counter={authors.length} />
        <Paragraph>
          <Title>
            Why weâ€™re suggesting these researches for your research project
          </Title>
          <br />
          <span
            dangerouslySetInnerHTML={{
              __html: sentence
            }}
          ></span>
        </Paragraph>
      </Sidebar>
      <CardsContainer>
        <Card authors={authors} />
      </CardsContainer>
    </MainSection>
  )
}

const Title = styled.span`
  display: inline-block;
  color: ${({ theme }) => theme.colors.primary[0]};
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5;
`

const Paragraph = styled(Text.Normal)`
  font-size: 1.125rem;
  line-height: 2.25;
  color: #242424;
  height: fit-content;
  max-height: 60vh;
  word-break: break-all;
  text-align: start;
  padding: 2em;
  overflow: auto;
  background-color: white;
  border-radius: 1.25em;
  border: 1px solid #c9d2ec;
`

const MainSection = styled.section`
  width: 100%;
  padding: 4em;
  display: grid;
  column-gap: 3em;
  grid-template-columns: 0.5fr 1fr;
  grid-auto-flow: column;
  background-color: ${({ theme }) => theme.colors.primary[0]};
`

const Sidebar = styled.div`
  grid-column: 1;
  display: flex;
  flex-direction: column;
  height: fit-content;
  gap: 1em;
  position: sticky;
  top: 2em;
`

const CardsContainer = styled.div`
  grid-column: 2;
  display: flex;
  flex-direction: column;
  gap: 1em;
  width: 100%;
`

export { SearchResults }
