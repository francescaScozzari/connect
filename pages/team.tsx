import React from 'react'
import styled from 'styled-components'
import type { NextPage } from 'next'

import { H2, Text } from '@/components/commons/Typography'
import { Button } from '@/components/commons/Button'
import { Navbar } from '@/components/team/Navbar'
import { Card } from '@/components/team/Card'
import { useAppSelector } from '@/store'

const Team: NextPage = () => {
  const { authors } = useAppSelector(state => state.search)

  return (
    <>
      <Navbar />
      <Container>
        <TeamSection>
          <H2.Normal>Team</H2.Normal>
          <Text.NormalBold>
            Abbiamo individuato per te un team multidisciplinare di 9 profili in
            linea con il tuo progetto di ricerca
          </Text.NormalBold>
          <CardsContainer>
            {authors?.map(author => {
              return (
                <Card
                  key={author.orcid}
                  fullName={author.fullName}
                  university={author.university}
                />
              )
            })}
          </CardsContainer>
          <SuggestionButton>Perch√® ti suggeriamo questo team</SuggestionButton>
        </TeamSection>
      </Container>
    </>
  )
}

const TeamSection = styled.section`
  display: flex;
  align-items: start;
  justify-content: center;
  flex-direction: column;
  gap: 1em;
`

const Container = styled.div`
  display: flex;
  padding: 2.5em;
  gap: 3em;
  min-height: calc(100vh - 80px);
  align-items: center;
  justify-content: start;
  flex-direction: column;
`

const CardsContainer = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  margin-top: 1.5em;
  gap: 1.5em;
`

const SuggestionButton = styled(Button)`
  align-self: end;
  background-color: black;
`

export default Team
