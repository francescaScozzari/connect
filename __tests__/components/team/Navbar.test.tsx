import React from 'react'
import { expect } from '@jest/globals'
import { screen } from '@testing-library/react'

import { Navbar } from '@/components/team/Navbar'
import { renderWithWrappers } from '@/__tests__/functions'

const setup = () => renderWithWrappers(<Navbar />)

describe('Navbar component', () => {
  test('Navbar renders correctly', () => {
    const container = setup()
    expect(container.firstChild).toMatchSnapshot()
  })
})
