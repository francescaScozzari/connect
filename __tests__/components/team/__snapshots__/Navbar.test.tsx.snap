// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Navbar renders correctly 1`] = `
.c4 {
  border: 2px solid black;
  box-sizing: border-box;
  border-radius: 0.5em;
  padding: 0.625em 2em;
  width: 25em;
  font-style: normal;
  font-size: 1.125rem;
}

.c4::placeholder {
  padding-left: 1.5em;
  font-size: 1.125rem;
  background: url("data:image/svg+xml,%3Csvg height='24' width='24' xmlns='http://www.w3.org/2000/svg'%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill:none;stroke:%23000;stroke-linecap:round;stroke-linejoin:round;stroke-width:2px%7D%3C/style%3E%3C/defs%3E%3Cg id='_21.search'%3E%3Ccircle class='cls-1' cx='9' cy='9' r='8'/%3E%3Cpath class='cls-1' d='m15 15 8 8'/%3E%3C/g%3E%3C/svg%3E") no-repeat 0.5em center;
}

.c3 {
  display: flex;
  flex-direction: column;
  margin-bottom: 1.5em;
}

.c3 p {
  font-size: 0.75em;
  margin-left: 0.625em;
  color: #BA1B23;
}

.c1 {
  margin: 0;
  text-transform: none;
}

.c2 {
  font-size: 24px;
  font-weight: 500;
}

.c0 {
  width: 100%;
  background-color: #F4F7F8;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 2em 3em;
}

.c0 a {
  color: black;
  text-decoration: none;
}

.c0 div {
  margin: 0 auto;
}

@media (max-width: 768px) {
  .c4 {
    font-size: 1em;
    width: 15em;
  }
}

@media (max-width: 768px) {
  .c0 div {
    display: none;
  }
}

<nav
  class="c0"
>
  <h3
    class="c1 c2"
  >
    <a
      href="/"
    >
      Lorem Ipsum
    </a>
  </h3>
  <div
    class="c3"
  >
    <input
      class="c4"
      placeholder=""
    />
  </div>
</nav>
`;
